canonical/vanilla-framework
# Document hero macro

## Done

- Adds documentation of the hero macro to the hero page.
- Adds some markup comments to the major blocks in the hero macro to make each element of the pattern more distinct in the HTML code snippets.
- Minor hero macro API changes per naming convention discussions

Note: this PR does not show the hero macro call. #5321 introduces the ability to view the Jinja macro call in the docs page.

Fixes [WD-14522](https://warthogs.atlassian.net/browse/WD-14522)

## QA

- Open [hero docs](https://vanilla-framework-5325.demos.haus/docs/patterns/hero)
- Review the macro documentation and verify you have a clear idea of how to use the macro.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
<img width="873" alt="Screenshot 2024-08-29 at 2 24 00 PM" src="https://github.com/user-attachments/assets/4547c2ed-3905-4583-b92c-a67b281f2808">

[WD-14522]: https://warthogs.atlassian.net/browse/WD-14522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
